src/components/coin-row/TransactionCoinRow.js

```diff
@@ -192,14 +189,22 @@ export default function TransactionCoinRow({ item, ...props }) {
         }
       );
     }
-  }, [accountAddress, contact, item, navigate]);
+  }, [accountAddress, contact, ens, item, navigate]);
 
   const mainnetAddress = useSelector(
     state =>
       state.data.accountAssetsData?.[`${item.address}_${item.network}`]
         ?.mainnet_address
   );
 
+  useEffect(() => {
+    const fetchEns = async () => {
+      const ensName = await fetchReverseRecord(contact?.address);
```

Comment:
since we're refactoring some of this, could we add ens names inside the contact object?

Reply:
i see we're using `contact?.ens` in other places, maybe we can do the same? if so this comment is out of date by now

Reply:
part of https://github.com/rainbow-me/rainbow/pull/3464/files#r951968013